### ISSUE TITLE
Detects unnecessary append

### ISSUE THREAD

**Issue by breml** [go-critic/go-critic#823]
Detects unnecessary append

Before:

```Go
x := make([]int,10)
x = append(x)
x = append(x[:a])
x = append(x[a:])
```

After:

```Go
x := make([]int,10)
x = x
x = x[:a]
x = x[a:]
```

Note: If the second argument in a call to `append` is omitted, append can be omitted entirely.

**Comment by quasilyte** [comment]
Pretty sure I've planned detecting `append(x)` calls before.

**Comment by quasilyte** [comment]
This will be implemented pretty soon. Thanks for the remainder.

**Comment by breml** [comment]
Sorry, currently I do not have time to implement this.
I stumbled over this in a code review (where it was not catched by go-critic) and I thought, this would be an easy thing to add to go-critic.

**Comment by quasilyte** [comment]
No problems.
I actually did a small investigation.
It does trigger in some projects:
https://github.com/moby/moby/pull/38888
https://github.com/flynn/flynn/pull/4512
https://github.com/future-architect/vuls/pull/785
https://github.com/hashicorp/vault/pull/6360
